app/views/shared/_flash_message.html.erb

```diff
@@ -1,19 +1,10 @@
-<% [:error, :success, :warning, :notice].each do |key| %>
-  <% if flash[key].is_a?(Array) %>
-    <% unless flash[key] -%>
-      <%= content_tag :div, nil, class: "#{key}", style: "display: none" %>
-    <% else %>
-      <%= content_tag :div, class: "#{key}" do
-        flash[key].each do |msg|
-          concat content_tag(:p, msg)
-        end
-      end %>
-    <% end %>
-  <% else %>
-    <% unless flash[key] %>
-      <%= content_tag :div, nil, class: "#{key}", style: "display: none" %>
-    <% else %>
-      <%= content_tag :div, flash[key], class: "#{key}" %>
-    <% end %>
+<% [:notice, :error, :success, :warning].each do |key| %>
+  <% unless flash[key].nil? %>
+    <% flash[key] = [flash[key]] unless flash[key].is_a?(Array)%>
```

Comment:
In general please use a space inside `<% ... %>` (so add one before the closing `%>` on this line. Also see below.